src/cdomains/arrayDomain.ml

```diff
@@ -693,55 +781,79 @@ module FlagConfiguredArrayDomain(Val: LatticeWithSmartOps) (Idx:IntDomain.Z):S w
 struct
   module P = PartitionedWithLength(Val)(Idx)
   module T = TrivialWithLength(Val)(Idx)
+  module U = Unroll(Val)(Idx)
 
   type idx = Idx.t
   type value = Val.t
 
-  include LatticeFlagHelper(P)(T)(struct
-      let msg = "FlagConfiguredArrayDomain received a value where not exactly one component is set"
-      let name = "FlagConfiguredArrayDomain"
-    end)
+  module K = struct
+    let msg = "FlagConfiguredArrayDomain received a value where not exactly one component is set"
+    let name = "FlagConfiguredArrayDomain"
+  end
+
+  let to_t = function
+  | (Some p, None, None) -> (Some p, None)
+  | (None, Some t, None) -> (None, Some (Some t, None))
+  | (None, None, Some u) -> (None, Some (None, Some u))
+  | _ -> failwith "FlagConfiguredArrayDomain received a value where not exactly one component is set"
+
+  module I = struct include LatticeFlagHelper (T) (U) (K) let name () = "" end
+  include LatticeFlagHelper (P) (I) (K)
+
+  let binop' ops ophs opks = binop ops (I.binop ophs opks)
```

Comment:
Why `ops`, `ophs`, and `opks`? These made sense for the struct domain with

~~~OCaml
  module S = Simple(Val)
  module HS = Sets(Val)
  module KS = KeyedSets(Val)
 ~~~
 
 Here, they should probably be named `opp` `opt` and `opu`.

Reply:
Fixed on 3c946da793bb674eff15d0316c9686ebf05e9301